### ISSUE TITLE
Authentication bug

### ISSUE THREAD

**Issue by NunuM** [AlexPikalov/cdrs#296]
Hi again @AlexPikalov ,

I was testing Cassandra 4.0~alpha2 version, and after setup one cluster with three nodes, I take this driver to play with it, but I could authenticate into the cluster, giving the error:

```
ERROR:<unknown>: Server error: CString { string: "Credential format error: username or password is empty or contains NUL(\\0) character" }
```

Digging up, I have used **strace** command, on production cluster that has the Cassandra 3, and on the new cluster with Cassandra 4

```
strace -xx -o strace-log -f -s 200 cqlsh -u mazikeen -p *********

# searching for the opcode AUTH RESPONSE
cat  strace-log | grep sendto | grep 0f

# I have remove the user and password from the following frames results 


#Cassandra4 Frame payload
\x04\x00\x00\x01\x0f\x00\x00\x00\x1c\x00\x00\x00\x18\x00 

#Cassandra3 Frame payload
\x04\x00\x00\x01\x0f\x00\x00\x00\x1c\x00\x00\x00\x18\x00
```

Then, I took the frame payload from this driver:

```
\x04\x00\x85\x85\x0f\x00\x00\x00\x20\x00\x00\x00\x1c\x00\x00\x00\x18\x00
```

Then, I dig the code, and notice that on file tcp_connection_pool.rs line 128, the body of the frame already is parsed into cbytes, that prefixes the size of the payload.
```
  128:  let auth_token_bytes = session_authenticator. get_auth_token().into_cbytes();
  129:  transport.borrow_mut().write(
            Frame::new_req_auth_response(auth_token_bytes)
                .into_cbytes()
                .as_slice(),
        )?;
```

In next line, when building the actual frame, the body is again passed into cbytes, prefixing a new size to body payload.

```
pub fn new_req_auth_response(bytes: Vec<u8>) -> Frame {
        let version = Version::Request;
        let flag = Flag::Ignore;
        let stream = rand::random::<u16>();
        let opcode = Opcode::AuthResponse;
        let body = BodyReqAuthResponse::new(CBytes::new(bytes));

        Frame {
            version: version,
            flags: vec![flag],
            stream: stream,
            opcode: opcode,
            body: body.into_cbytes(),
            // for request frames it's always None
            tracing_id: None,
            warnings: vec![],
        }
    }
```  

My quick fix, was to:

```
pub fn new_req_auth_response(bytes: Vec<u8>) -> Frame {
        let version = Version::Request;
        let flag = Flag::Ignore;
        let stream = rand::random::<u16>();
        let opcode = Opcode::AuthResponse;
        //let body = BodyReqAuthResponse::new(CBytes::new(bytes));

        Frame {
            version: version,
            flags: vec![flag],
            stream: stream,
            opcode: opcode,
            body: bytes,
            // for request frames it's always None
            tracing_id: None,
            warnings: vec![],
        }
    }
```

And test against Cassandra3 and 4 and have worked in both. 

I have not have read the full protocol spec, and don't know why is working on Cassandra3 with the current driver implementation, but the python client for V3 is not sending the extra 4 bytes, if there is a special reason let me know, otherwise, I can make a PR with your help.

Thanks

**Comment by AlexPikalov** [comment]
Hi @NunuM 
Thanks for submitting the issue and for details explanation. I'll need to look closer on that.
Please give me a day or something

**Comment by NunuM** [comment]
@AlexPikalov my pleasure. Yes, take your time 👍

**Comment by AlexPikalov** [comment]
@NunuM

Looks like you're absolutely right about extra 4 bytes.

According to the [spec](https://github.com/apache/cassandra/blob/trunk/doc/native_protocol_v4.spec#L294):

> The body of this message is a single [bytes] token.

```rust
let auth_token_bytes = session_authenticator. get_auth_token().into_cbytes();
```

It's already a Cassandra bytes (4 bytes of length + following bytes) so no need to convert them again like it's done here.

Could you please submit a PR?

**Comment by AlexPikalov** [comment]
Only one thing. I like the original idea to have a separate structure for each type of body. So if we can agree that [this](https://github.com/AlexPikalov/cdrs/blob/master/src/frame/frame_auth_response.rs#L8) already contains bytes then we can modify `new_req_auth_response` in following way:

```rust
pub fn new_req_auth_response(token_bytes: CBytes) -> Frame {
        let version = Version::Request;
        let flag = Flag::Ignore;
        let stream = rand::random::<u16>();
        let opcode = Opcode::AuthResponse;
        let body = BodyReqAuthResponse::new(token_bytes);

        Frame {
            version: version,
            flags: vec![flag],
            stream: stream,
            opcode: opcode,
            body: body.into_cbytes(),
            // for request frames it's always None
            tracing_id: None,
            warnings: vec![],
        }
    }
```
while having [here](https://github.com/AlexPikalov/cdrs/blob/master/src/cluster/tcp_connection_pool.rs#L128) just

```rust
let auth_token_bytes = session_authenticator.get_auth_token();
```
Does it make sense?

**Comment by AlexPikalov** [comment]
Fixed in #299
